alacritty_terminal/src/term/mod.rs

```diff
@@ -2118,10 +2117,23 @@ impl ansi::Handler for Term {
     }
 
     #[inline]
-    fn set_scrolling_region(&mut self, region: Range<Line>) {
-        trace!("Setting scrolling region: {:?}", region);
-        self.scroll_region.start = min(region.start, self.grid.num_lines());
-        self.scroll_region.end = min(region.end, self.grid.num_lines());
+    fn set_scrolling_region(&mut self, top: usize, bottom: usize) {
+        if top >= bottom {
+            debug!("Invalid scroll region: ({};{})", top, bottom);
+            return;
+        }
+
+        // Bottom should be included in the range, but range end is not
+        // usually included. One option would be to use an inclusive
+        // range, but instead we just let the open range end be 1
+        // higher.
+        let start = Line(top - 1);
+        let end = Line(bottom);
+
+        trace!("Setting scrolling region: {}x{}", start, end);
```

Comment:
I wouldn't use a `x` to denote a range. To be `x` means width x height, or age x gender or generally any two separate dimensions. While `..` or `...` (depending who you ask) is better for a range. However since we're in Rust I'd stick with `..` meaning half-open. If you're trying to avoid that then maybe `x-y` would even be better.

Sorry I love being pedantic.